data/pigui/modules/saveloadgame.lua

```diff
@@ -120,20 +113,36 @@ local function closeAndLoadOrSave()
 end
 
 ui.saveLoadWindow = ModalWindow.New("LoadGame", function()
-	local mode = ui.saveLoadWindow.mode == 'SAVE' and lui.SAVE or lui.LOAD
-	optionTextButton(mode, nil, selectedSave ~= nil and selectedSave ~= '' and saveIsValid, closeAndLoadOrSave)
-	ui.sameLine()
-	optionTextButton(lui.CANCEL, nil, true, closeAndClearCache)
+	local saving = ui.saveLoadWindow.mode == "SAVE"
+	local other_height = optionButtonSize.y + ui.getItemSpacing().y * 2 + ui.getWindowPadding().y * 2
+	ui.child("savefiles", Vector2(-1, winSize.y - other_height), function()
+		showSaveFiles()
+	end)
 
-	if ui.saveLoadWindow.mode == "SAVE" then
+	ui.separator()
+
+	local txt_width = winSize.x - (ui.getWindowPadding().x + optionButtonSize.x + ui.getItemSpacing().x) * 2
+	if saving then
+		-- for vertical center alignment
+		local txt_hshift = math.max(0, (optionButtonSize.y - ui.getFrameHeight()) / 2)
```

Comment:
There is a function `ui.alignTextToButtonPadding` that performs much the same function; it could be trivially extended to take an explicit button size parameter and avoid manually doing the math here. Using that function makes the code *much* clearer semantically as well.